Doc/library/tkinter.filedialog.rst

```diff
@@ -0,0 +1,142 @@
+:mod:`tkinter.filedialog` --- File selection dialogs
+====================================================
+
+.. module:: tkinter.filedialog
+   :platform: Tk
+   :synopsis: Dialog classes for file selection
+
+**Source code:** :source:`Lib/tkinter/filedialog.py`
+
+--------------
+
+The :mod:`tkinter.filedialog` module provides classes and factory methods that
+facilitate file/directory selection.
+
+
+.. FIXME Methods that should (or should not) be documented should be added in
+   (or removed) as appropriate
+
+.. class:: Directory(self, master=None, **options)
+
+   Open dialog prompting user to select a directory
+
+
+.. class:: FileDialog(self, master, title=None)
+
+   Standard file selection dialog
+
+    .. function:: cancel_command(self, event=None)
+
+       Exit the file selection dialog
+
+    .. function:: dirs_double_event(self, event)
+
+       Event handler for double-click event on directory
+
+    .. function:: dirs_select_event(self, event)
+
+       Event handler for click event on directory
+
+    .. function:: files_double_event(self, event)
+
+       Event handler for double-click event on file
+
+    .. function:: files_select_event(self, event)
+
+       Event handler for double-click event on file
+
+    .. function:: filter_command(self, event=None)
+
+       Filters the files by directory
+
+    .. function:: get_filter(self)
+
+       Retrieve the file filter currently in use
+
+    .. function:: get_selection(self)
+
+       Retrieve the currently selected item
+
+    .. function:: go(self, dir_or_file=os.curdir, pattern="*", default="", key=None)
+
+        Displays dialog and starts event loop
+
+    .. function:: ok_event(self, event)
+
+       Exit dialog returning current selection
+
+    .. function:: quit(self, how=None)
+
+       Exit dialog returning filename, if any
+
+    .. function:: set_filter(self, dir, pat)
+
+        Set the file filter
+
+    .. function:: set_selection(self, file)
+
+
+.. class:: LoadFileDialog(self, master, title=None)
+
+   File selection dialog that checks that the file exists
+
+    .. function:: ok_command(self)
+
+       Exit dialog returning current selection
+
+
+.. class:: Open(self, master=None, **options)
+
+   Native tk dialog for retrieving filename to open
+
+
+.. class:: SaveAs(self, master=None, **options)
+
+   Native tk dialog for retrieving filename for saving
+
+
+.. class:: SaveFileDialog(self, master, title=None)
```

Comment:
I am inclined to leave these two separate - primarily because unlike the *Open* and *SaveAs* classes they are not native dialogs (with regard to look-and-feel). 
As such in my mind they exist for the purpose of custom behaviours, so grouping them in the documentation would feel counter-intuitive to that approach